### DISCUSSION TITLE
java-library - test code:  No Embedded Server

### DISCUSSION THREAD

**Discussion by waveskimmer** [micronaut-projects/micronaut-core#11201] [Q&A]
I've had this unresolved problem where when writing a library (e.g., using `io.micronaut.minimal.library` plugin), I'm unable to get ahold of the `EmbeddedServer`.    

That is: 

`            EmbeddedServer embeddedServer = ApplicationContext.run(EmbeddedServer.class);`

errors. 

I generated a quick app with the launcher to demonstrate:   https://github.com/waveskimmer/micronaut-embedded-server

**Comment by yawkat** [comment]
You need a http server implementation as a dependency
